torchtune/modules/transformer.py

```diff
@@ -261,22 +261,27 @@ def forward(
 
         # A mask of tokens (x) with no encoder_input
         skip_mask = self._skip_mask(encoder_mask)
+        if encoder_mask is not None:
+            # TODO: remove after PyTorch 2.5 is released
+            # This unmasks the skipped rows to avoid NaNs in SPDA Softmax backward
```

Comment:
```suggestion
            # This unmasks the skipped rows to avoid NaNs in SDPA Softmax backward
```

Reply:
Also is this todo only related to encoder_mask? Or can we remove the usage of skip_mask altogether when 2.5 is released? I had assumed it was the latter

Reply:
We still need skip_mask, we just don't need this extra step of updating the encoder_mask. The update would make skip_mask optional as you wouldn't have to worry about NaN's but you still get different behavior as you're not masking out the ffwd or output in attention.